lib/logproto/logproto-text-server.c

```diff
@@ -384,7 +218,7 @@ log_proto_text_server_fetch_from_buffer(LogProtoBufferedServer *s, const guchar
         }
       else
         {
-          log_proto_text_server_split_buffer(self, state, buffer_start, buffer_bytes);
+          log_proto_buffered_server_split_buffer(&self->super, state, buffer_start, buffer_bytes);
           goto exit;
```

Comment:
Thanks, the last changes look good.

One more idea: what if we pushed the the invocation had 3 different returns from "fetch_from_buffer"
* We have a line (current TRUE value)
* we don't have a line but the buffer is full and cannot be split
* We don't have a line but can split.
* 
Now as I think of it the last two cases don't even need separate returns. The decision whether we can split can be decided at the buffered-server level.

So maybe you could copy the remainder of the text_server_fetch_from_buf() to the call site.

Just an idea.

Reply:
I'll try that, thanks.

One thing I noticed is that we currently split the buffer too often: for example, when a message is received from N partial read operations, each attempt will trigger a `memmove()` operation. I believe we only need to split our buffer when we're about to reach the end of it.

If I'm right about this, I could easily remove the split logic from the text server completely (moving it to the buffered server), and we may also increase performance a bit.

Reply:
Done.

Reply:
Hmm, I might be wrong about the performance impact of my last commit, because I've introduced more `read()` and `iconv()` calls on smaller chunks of data to save a few `memmove()`s. I'm dropping that commit.